src/memory/sqlite_store.rs

```diff
@@ -48,42 +49,70 @@ impl<Address, Word> Debug for SqliteMemory<Address, Word> {
 
 // The following settings are used to improve performance:
 // - journal_mode = WAL : WAL journaling is faster than the default DELETE mode.
-// - synchronous = NORMAL: Reduces disk I/O by only calling fsync() at critical moments rather
-//   than after every transaction (FULL mode); this does not compromise data integrity.
-const CREATE_TABLE_SCRIPT: &str = "
+// - synchronous = NORMAL: Reduces disk I/O by only calling fsync() at critical
+//   moments rather than after every transaction (FULL mode); this does not
+//   compromise data integrity.
+fn create_table_script(table_name: &str) -> String {
+    format!(
+        "
 PRAGMA synchronous = NORMAL;
-PRAGMA journal_mode = WAL;
-CREATE TABLE IF NOT EXISTS memory (
+CREATE TABLE IF NOT EXISTS {} (
     a BLOB PRIMARY KEY,
     w BLOB NOT NULL
-);";
+);",
+        table_name
+    )
+}
 
 impl<Address, Word> SqliteMemory<Address, Word> {
-    /// Connects to a known DB using the given path.
+    /// Builds a pool connected to a known DB (using the given path) and creates
+    /// the `findex_memory` table.
     ///
     /// # Arguments
     ///
     /// * `path` - The path to the sqlite3 database file.
     pub async fn connect(path: &str) -> Result<Self, SqliteMemoryError> {
-        // This pool connections number defaults to the number of logical CPUs
-        // of the current system.
-        let pool = PoolBuilder::new().path(path).open().await?;
+        // This pool connections number defaults to the number of logical CPUs of the
+        // current system.
+        let pool = PoolBuilder::new()
+            .journal_mode(async_sqlite::JournalMode::Wal)
+            .path(path)
+            .open()
+            .await?;
 
-        pool.conn(move |conn| conn.execute_batch(CREATE_TABLE_SCRIPT))
+        pool.conn(move |conn| conn.execute_batch(&create_table_script(FINDEX_TABLE_NAME)))
             .await?;
 
         Ok(Self {
             pool,
             _marker: PhantomData,
         })
     }
+
+    /// Creates the memory's table using a connection from the given pool.
```

Comment:
I don't feel the current doc describes what the functions does. Maybe:
```suggestion
    /// Returns an `SqliteMemory` instance storing data in a table with the given name
    /// and connecting to the DB using connections from the given pool.
```